tests/flow/test_pending_queries_limit.py

```diff
@@ -11,7 +11,11 @@
 error_encountered = False
 
 GRAPH_NAME = "max_pending_queries"
-SLOW_QUERY = "UNWIND range (0, 100000) AS x WITH x WHERE (x / 2) = 50  RETURN x"
+# on machines with many cores on some runs the pace at which the queries been executed by the workers 
+# is greater than the insertion to the queues. 
+# So for the whole test there was at most one job in the queue which failed:
+# test_03_overflow_with_limit. Increasing the execution time of the query resolved this.
+SLOW_QUERY = "UNWIND range (0, 5000000) AS x WITH x WHERE (x / 2) = 50  RETURN x"
```

Comment:
Wouldn't it be better to issue additional concurrent queries,
say if we're running on a 32 core machine, make sure we're sending 128 concurrent queries?

Reply:
It will not solve the problem, since the rate at which the threads are handling the work is greater the the reate at which RedisServer which is serial pushes the work.